docker/docker.sh

```diff
@@ -37,9 +37,26 @@ function install_docker() {
   apt-get install -y docker-ce="${DOCKER_VERSION}"
 }
 
+function configure_gcr() {
+  VERSION=1.5.0
+  OS=linux
+  ARCH=amd64
+
+  # this standalone method is recommended here:
+  # https://cloud.google.com/container-registry/docs/advanced-authentication#standalone_docker_credential_helper
+  curl -fsSL "https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v${VERSION}/docker-credential-gcr_${OS}_${ARCH}-${VERSION}.tar.gz" \
+    | tar xz --to-stdout ./docker-credential-gcr \
+    > /usr/bin/docker-credential-gcr && chmod +x /usr/bin/docker-credential-gcr
+  docker-credential-gcr configure-docker
+  su yarn --command "docker-credential-gcr configure-docker"
```

Comment:
I don't see why to run this as YARN. My (limited) understanding is this makes all Docker users able to read from GCR.

Does that not work?

Reply:
You would think so, right?  I actually went back to double check this when you asked, because that was my intuition as well.  However, this command configures docker on a per-user basis.  Unfortunately I have not been able to find any way to configure docker to use GCR globally.  Root and yarn are the two static users that I could see using docker, and I run as yarn since the configuration function seems to indicate that yarn is the only other user added to the docker group.  I will add a comment that makes this explicit.

Reply:
A viable future change may be to let the user pass docker-enabled users via a metadata variable, and generalize beyond yarn this way.